as-pu-conv/as_pu_conv/main.py

```diff
@@ -86,12 +86,12 @@ def check_connected(self, psname):
     def process(self):
         """Process all write requests in queue and does a BBB scan."""
         t0_ = _time.time()
-        if self._dequethread:  # Not None and not empty
-            status = self._dequethread.process()
+        if self._queuethread is not None and self._queuethread.qsize():
```

Comment:
You don't need to check for the `qsize()` in your `if`'s. You can still use the old way.

Reply:
done!

Reply:
In fact, as we talked offline, I removed the implementation of this suggestion because the sentences are different, once bool(Queue()) is equal to True.